usl_pipeline/cloud_functions/main.py

```diff
@@ -291,6 +293,83 @@ def delete_flood_scenario_metadata(cloud_event: functions_framework.CloudEvent)
 
 
+@functions_framework.cloud_event
+@_retry_and_report_errors()
+def build_label_matrix(cloud_event: functions_framework.CloudEvent) -> None:
+    """Builds a label matrix when a set of simulation output files is uploaded.
+
+    This function is triggered when files containing simulation data are uploaded to
+    GCS. It produces a label matrix for the sim data and writes that label matrix to
+    another GCS bucket for eventual use in model training and prediction.
+    """
+    _build_label_matrix(
+        cloud_event.data["bucket"],
+        cloud_event.data["name"],
+        cloud_storage.LABEL_CHUNKS_BUCKET,
+    )
+
+
+def _build_label_matrix(bucket_name: str, chunk_name: str, output_bucket: str) -> None:
+    storage_client = storage.Client()
+    bucket = storage_client.bucket(bucket_name)
+    chunk_blob = bucket.blob(chunk_name)
+
+    with chunk_blob.open("rb") as chunk:
+        # Heat (WRF) - treat one WRF output file as one chunk
+        if re.search(file_names.WRF_DOMAIN3_NC_REGEX, chunk_name):
+            label_matrix = _build_wrf_label_matrix(chunk)
+
+            label_file_name = chunk_name + ".npy"
+            label_blob = storage_client.bucket(output_bucket).blob(str(label_file_name))
+
+            _write_as_npy(label_blob, label_matrix)
+            _write_wrf_label_chunk_metastore_entry(chunk_blob, label_blob)
+        else:
+            raise ValueError(f"Unexpected file {chunk_name}")
+
+
+def _build_wrf_label_matrix(fd: IO[bytes]) -> NDArray:
+    label_components = []
+    with netCDF4.Dataset("ncfile", mode="r", memory=fd.read()) as nc:
+        # Variable names are listed here according to the formatting
+        # specified by wrf-python documentation
+        # https://wrf-python.readthedocs.io/en/latest/user_api/generated/wrf.getvar.html#wrf-getvar
+        vars_to_derive = ["rh2", "T2", "wspd_wdir10"]
+        for var in vars_to_derive:
+            if var == "wspd_wdir10":
+                # TODO: Split wind dir to sin/cos components
+                wspd10, wdir10 = wrf.getvar(nc, "wspd_wdir10")
+                label_components.append(numpy.swapaxes(wspd10, 0, 1))
+                label_components.append(numpy.swapaxes(wdir10, 0, 1))
+            else:
+                label = wrf.getvar(nc, var)
+                # Ensure order of dimension axes are: (west_east, south_north)
+                label_components.append(numpy.swapaxes(label, 0, 1))
+
+    labels_matrix = numpy.dstack(label_components)
+    return labels_matrix
+
+
+def _write_wrf_label_chunk_metastore_entry(
+    chunk_blob: storage.Blob, label_blob: storage.Blob
+) -> None:
+    """Updates the metastore with new information for the given wrf chunks.
+
+    Args:
+      chunk_blob: The GCS blob containing the raw chunk.
+      label_blob: The GCS blob containing the label tensor for the chunk.
+    """
+    db = firestore.Client()
+    study_area_name = _parse_chunk_path(chunk_blob.name)[0]
+
+    metastore.SimulationLabelChunk(
```

Comment:
@waltaskew Do we want a WRF-specific simulation label chunk object here?

Reply:
ah, yes, I think we'll want to mirror
https://github.com/UrbanSystemsLab/climateiq-cnn/pull/96
in that we're dividing not dividing these up spatially as we do for flood

Reply:
Ack - I can throw that in this PR!